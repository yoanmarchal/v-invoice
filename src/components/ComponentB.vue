<template>
  <div class="component-b">
    <input type="text" v-model="value">
    <button @click="onClick">Click me</button>
  </div>
</template>

<script>
export default {
  data () {
    return {
      value: 'Vue'
    }
  },
  methods: {
    onClick () {
      this.$emit('click', this.value)
    }
  }
}
</script>
