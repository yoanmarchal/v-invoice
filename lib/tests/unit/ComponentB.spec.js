import { mount } from '@vue/test-utils'
import ComponentB from '@/components/ComponentB'

describe('ComponentB.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = mount(ComponentB)
  })

  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('triggers a click event when the button is clicked', () => {
    wrapper.setData({ value: 'Vue2' })
    wrapper.find('button').trigger('click')
    expect(wrapper.emitted().click).toBeTruthy()
    expect(wrapper.emitted().click[0]).toEqual(['Vue2'])
  })
})
