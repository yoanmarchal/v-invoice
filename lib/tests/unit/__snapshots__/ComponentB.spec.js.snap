// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentB.vue has the expected html structure 1`] = `
<div
  class="component-b"
>
  <input
    type="text"
  />
   
  <button>
    Click me
  </button>
</div>
`;
